cvat/apps/dataset_manager/views.py

```diff
@@ -32,78 +36,123 @@ def log_exception(logger=None, exc_info=True):
             (_MODULE_NAME, current_function_name(2)),
         exc_info=exc_info)
 
+EXPORT_CACHE_DIR_NAME = 'export_cache'
 
 def get_export_cache_dir(db_instance):
     base_dir = osp.abspath(db_instance.get_dirname())
 
     if osp.isdir(base_dir):
-        return osp.join(base_dir, 'export_cache')
+        return osp.join(base_dir, EXPORT_CACHE_DIR_NAME)
     else:
-        raise FileNotFoundError('{} dir {} does not exist'.format(db_instance.__class__.__name__, base_dir))
+        raise FileNotFoundError(
+            '{} dir {} does not exist'.format(db_instance.__class__.__name__, base_dir)
+        )
 
-DEFAULT_CACHE_TTL = timedelta(hours=10)
-TASK_CACHE_TTL = DEFAULT_CACHE_TTL
+DEFAULT_CACHE_TTL = timedelta(seconds=settings.DATASET_CACHE_TTL)
 PROJECT_CACHE_TTL = DEFAULT_CACHE_TTL / 3
+TASK_CACHE_TTL = DEFAULT_CACHE_TTL
 JOB_CACHE_TTL = DEFAULT_CACHE_TTL
 
+EXPORT_CACHE_LOCK_TIMEOUT = timedelta(seconds=settings.DATASET_CACHE_LOCK_TIMEOUT)
+
+def get_export_cache_ttl(db_instance: str | Project | Task | Job) -> int:
+    TTL_CONSTS = {
+        'project': PROJECT_CACHE_TTL,
+        'task': TASK_CACHE_TTL,
+        'job': JOB_CACHE_TTL,
+    }
+
+    if not isinstance(db_instance, str):
+        db_instance = db_instance.__class__.__name__.lower()
+
+    return TTL_CONSTS[db_instance].total_seconds()
+
+def get_file_instance_timestamp(file_path: str) -> float:
+    match = re.search(r'instance(\d+\.\d+)', osp.basename(file_path))
```

Comment:
I'm not sure this is robust enough. The file name includes many components, and in the future, might include even more; can you really be sure that the string `instanceM.N` will never occur at an unexpected position within it?

IMO, to make this work in a robust matter, you need to create two functions: one to create a file name from components (and ensure it's unambiguous), and another to parse an entire filename into components. If you place them side by side, that will at least give some assurance that whoever updates one will update the other.

(Even better would be to restructure the code to make parsing the file name unnecessary, but I haven't read far enough to know if it's possible.)

Reply:
> I'm not sure this is robust enough. The file name includes many components, and in the future, might include even more; can you really be sure that the string instanceM.N will never occur at an unexpected position within it?

I suppose, some tests should fail if it happens.

> IMO, to make this work in a robust matter, you need to create two functions: one to create a file name from components (and ensure it's unambiguous), and another to parse an entire filename into components. If you place them side by side, that will at least give some assurance that whoever updates one will update the other.

Ok, I'll check this variant.

> Even better would be to restructure the code to make parsing the file name unnecessary

This could be changed for a value in a side file, file tags, redis or the main db, but I decided to keep this information closer to the file to avoid potential out-of-sync problems. I'd like to use file creation time and update time, but there are issues with these values on different platforms. On linux, you can't simply reach file creation time. These values are also dependent on the filesystem options. Thus, using filename seems to be the most convenient option. Additionally, it automatically resolved the creation-removal races (so the only reading-removal race is left).

Reply:
> I suppose, some tests should fail if it happens.

Not necessarily. It could happen only with particular user inputs, which might not be tested.

Reply:
> This could be changed for a value in a side file, file tags, redis or the main db, but I decided to keep this information closer to the file to avoid potential out-of-sync problems.

Yeah, that's fine, I was just thinking out loud. 🙂 This does seem to be the most straightforward implementation, we just need to make sure to implement parsing in a robust way.

Reply:
> > IMO, to make this work in a robust matter, you need to create two functions: one to create a file name from components (and ensure it's unambiguous), and another to parse an entire filename into components. If you place them side by side, that will at least give some assurance that whoever updates one will update the other.

> Ok, I'll check this variant.

Added 2 functions to build and split export filenames.

> > I'm not sure this is robust enough. The file name includes many components, and in the future, might include even more; can you really be sure that the string instanceM.N will never occur at an unexpected position within it?

> I suppose, some tests should fail if it happens.

I've improved the parsing a little.